bin/synapse_index.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+import hashlib
+import os
+import re
+import sys
+
+import synapseclient
+from synapseclient.models import File
+
+
+# TODO: swap this for `md5_for_file_hex` in synapseclient V4.2.0
+def compute_md5_checksum(file: str) -> str:
+    """Computes the hexadecimal MD5 checksum of a file.
+    Args:
+        file: Name of the file.
```

Comment:
Nit: slightly confusing for this and the clean function to take a file argument. But they are slightly different things.

This function is really the path of the file. The clean is the path or the name of the file.

Reply:
You're right that is a little confusing. But also, both take the path to the file. The cleaning function then gets the base name of the file to clean up. I'll clarify in the docstrings/variable names